sql/core/src/test/resources/sql-tests/results/literals.sql.out

```diff
@@ -442,9 +442,10 @@ struct<>
 -- !query output
 org.apache.spark.sql.catalyst.parser.ParseException
 {
-  "errorClass" : "_LEGACY_ERROR_TEMP_0021",
+  "errorClass" : "UNSUPPORTED_TYPED_LITERAL",
   "messageParameters" : {
-    "valueType" : "GEO"
+    "supportedTypes" : "\"DATE\", \"TIMESTAMP_NTZ\", \"TIMESTAMP_LTZ\", \"TIMESTAMP\", \"INTERVAL\", \"X\"",
```

Comment:
isn't it a constant list? does it need to be a parameter?

Reply:
It depends on implementation, and we might support more in the future. Having the list in the source code can give the following benefits:
1. More likely, devs will not forget to update the list in source code.
2. If the list will be in the JSON file, tech editors might forget to edit it or accidentally modify it.
3. The list of parameters might be formatted in different ways ( as a foldable or drop-down list) by frontend tools.

Reply:
I think it should be a parameter, unless `UNSUPPORTED_TYPED_LITERAL` will only be used in this scenario